src/alchemlyb/tests/test_workflow.py

```diff
@@ -0,0 +1,32 @@
+import pytest
+from alchemlyb.workflows import base
+import pandas as pd
+from unittest.mock import patch
+import os
+
+class Test_automatic_base():
+    @staticmethod
+    @pytest.fixture(scope='class')
+    def workflow():
+        workflow = base.WorkflowBase()
+        workflow.run()
+        yield workflow
+
+    # def test_write(self, workflow, tmpdir):
```

Comment:
@orbeckst Thanks for the suggestion of using pytest.patch and tmpdir. However, I found it a bit difficult to use it in this case.
I wish to change the workflow.out to the temporary directory using patch and then save the result to the temporary directory, where workflow is a pytest fixture. I wonder what is the best way of doing this? Thank you.

Reply:
I would configure the output directory when you set up the fixture. If there's a kwarg then pass the tmpdir to the constructor. Otherwise, do something like
```python
workflow.out = tmpdir.strpath
```
although they now recommend using [tmp_path](https://docs.pytest.org/en/7.0.x/how-to/tmp_path.html) instead.

Normally I'd say use
```python
@pytest.fixture
def def workflow(tmp_path):
    outdir = tmp_path / "out"   # this out subdir is really optional, you can also just use tmp_path
    outdir.mkdir()              # and then you don't have to make the subdir either
    workflow = base.WorkflowBase(out=str(outdir))
    return workflow.run()
```
but that won't work with class-scoped fixtures.  There's a way to create these directories with [tmp_path_factory](https://docs.pytest.org/en/7.0.x/how-to/tmp_path.html#the-tmp-path-factory-fixture). I *think* you'd use it like this:
```python
@pytest.fixture(scope="session")
def def workflow(tmp_path_factory):
    outdir = tmp_path_factory.mktemp("out")
    workflow = base.WorkflowBase(out=str(outdir))
    return workflow.run()
```
According to the docs, it will create a session-scoped fixture. Therefore, you'd just put this fixture into the module instead of the class. I don't think that this would be a problem.